homeassistant/core.py

```diff
@@ -175,6 +178,12 @@ def is_callback(func: Callable[..., Any]) -> bool:
     return getattr(func, "_hass_callback", False) is True
 
 
+@callback
+def async_get_hass() -> HomeAssistant:
+    """Return the HomeAssistant instance."""
```

Comment:
Please add the expectations when it's ok to use this function in the docstring body.

Reply:
Updated to explain we still prefer passing `hass` around.

Reply:
Should we mention that it's only available after creating `hass` during bootstrap?

Reply:
Yes it would be good to explain the limitations. It's possible to call `asyng_get_hass` in scripts too eevn though they don't use the bootstrapping code.
How to explain it in a concise way? Maybe we can just mention that the function can be called if a `HomeAssistant` instance has been created and will throw if it hasn't?

Reply:
Sounds good.